src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -722,6 +727,13 @@ def cleanup_nodes(self) -> bool:
                 else:
                     to_reimage.append(node)
 
+        idle_nodes_ids = NodeHeartbeat.get_dead_nodes(
+            [node_id for node_id in azure_nodes.keys()], datetime.timedelta(hours=1)
```

Comment:
This call seems potentially expensive. Do we have a sense of how long we expect it to take? Should this be its own scheduled task?

Reply:
It not more expensive than the existing calls to the database. For reference 
- the target throughput is [2000 entity per sec](https://docs.microsoft.com/en-us/azure/storage/tables/scalability-targets#scale-targets-for-table-storage) assuming a 1-KiB entity size. 
- the node heartbeat record is about 258 bytes using these [numbers](https://docs.microsoft.com/en-us/rest/api/storageservices/Understanding-the-Table-Service-Data-Model?redirectedfrom=MSDN#property-types)
- the maximum number nodes under a scaleset is [1000](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits#virtual-machine-scale-sets-limits). 
I think we can expect a query time well under a seconds in the worst case.